boot/bootstate20.go

```diff
@@ -720,3 +720,44 @@ func trustedAssetsBootState(dev Device) *bootState20BootAssets {
 		dev: dev,
 	}
 }
+
+// bootState20CommandLine implements the successfulBootState interface for
+// kernel command line
+type bootState20CommandLine struct {
+	dev Device
+}
+
+func (ba20 *bootState20CommandLine) markSuccessful(update bootStateUpdate) (bootStateUpdate, error) {
+	u20, err := toBootStateUpdate20(update)
+	if err != nil {
+		return nil, err
+	}
+	if len(u20.modeenv.CurrentTrustedBootAssets) == 0 && len(u20.modeenv.CurrentTrustedRecoveryBootAssets) == 0 {
+		// XXX: does this change when we expose the ability to add
+		// things to the command line?
+
+		// not using trusted boot assets, bootloader config is not
+		// managed and command line can be manipulated externally
+		return update, nil
+	}
+
+	var newM *Modeenv
+	if len(u20.modeenv.CurrentKernelCommandLines) == 0 {
```

Comment:
maybe this if should be inside a single observe* helper that takes this decision inside?

Reply:
moved to a single func